### PR TITLE
[JENKINS-39336] Add links to  items for activities

### PR DESCRIPTION
**Pull request by imeredith**
# Description

See [JENKINS-39336](https://issues.jenkins-ci.org/browse/JENKINS-39336).

Adds links to all columns so that right clicking works. 

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees 

Adding links to all items in the row allows the item to be
opened in a new tab with right click.

### CODE REVIEW

**Comment by i386** [comment]
@imeredith this looks visually the same as before right?

**Comment by i386** [comment]
Remove the text-decoaration in CSS and then this PR is good to go 💯

**Comment by michaelneale** [comment]
functionally LGTM. @cliffmeyers do you want to comment on the <RunCol> etc (vs a <td> that it was in the past?)

**Comment by michaelneale** [comment]
:bee: why not. Do it. ATH passes.

File: blueocean-dashboard/src/main/js/components/Branches.jsx [hunk 1]
```diff
@@ -49,16 +56,16 @@ export default class Branches extends Component {
 
         return (
             <tr key={cleanBranchName} onClick={open} id={`${cleanBranchName}-${id}`} >
```
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/components/Branches.jsx]: Should the onClick handler be removed perhaps?

File: blueocean-dashboard/src/main/js/components/Branches.jsx [hunk 2]
```diff
@@ -49,16 +56,16 @@ export default class Branches extends Component {
...
-                <td><CommitHash commitId={commitId} /></td>
-                <td>{msg || '-'}</td>
-                <td><ReadableDate date={endTime} liveUpdate /></td>
+                </BranchCol>
+                <BranchCol>{cleanBranchName}</BranchCol>
+                <BranchCol><CommitHash commitId={commitId} /></BranchCol>
+                <BranchCol>{msg || '-'}</BranchCol>
+                <BranchCol><ReadableDate date={endTime} liveUpdate /></BranchCol>
                 { /* suppress all click events from extension points */ }
                 <td className="actions" onClick={(event) => stopProp(event)}>
```
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/components/Branches.jsx]: If the `onClick` handler is removed from the TR then the I believe this `onClick` can also be killed. To test it, try favoriting a branch and verify that Run Detail doesn't open.

**Comment by cliffmeyers** [comment]
Oops, I see this was merged before I could comment. Well, if @imeredith wants to respond to the feedback that would be nice. :)

**Comment by i386** [comment]
@cliffmeyers maybe open up another ticket and go from there?

**Pull request merged**